### PR TITLE
add ability to create collective on a social token

### PR DESCRIPTION
**Pull request by ThornM9**

### CODE REVIEW

File: packages/spl-token-collective/src/index.ts [hunk 1]
```diff
@@ -639,26 +642,48 @@ export class SplTokenCollective extends AnchorSdk<SplTokenCollectiveIDL> {
...
-        },
-        {
-          accounts: {
-            collective,
-            mint: mint!,
-            mintAuthority: mintAuthority!,
-            payer,
-            systemProgram: SystemProgram.programId,
-            rent: SYSVAR_RENT_PUBKEY,
+    if (socialTokenRef) {
```
Comment (ChewingGlass) [inline packages/spl-token-collective/src/index.ts:645]: You should be able to derive the socialTokenRef from the mint. So even if they don't pass this variable, you should still check to see if the token is a social token.

File: packages/spl-token-collective/src/index.ts [hunk 2]
```diff
@@ -571,6 +573,7 @@ export class SplTokenCollective extends AnchorSdk<SplTokenCollectiveIDL> {
     config,
     bonding,
     metadata,
+    socialTokenRef,
```
Comment (ChewingGlass) [inline packages/spl-token-collective/src/index.ts:576]: maybe just tokenRef, we never clarify social

File: packages/spl-utils/src/splToken.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export async function createMint(
   if (authority === undefined) {
     authority = provider.wallet.publicKey;
   }
-  const mint = new Account();
+  const mint = Keypair.generate();
```
Comment (ChewingGlass) [inline packages/spl-utils/src/splToken.ts:17]: What happened here?

File: programs/spl-token-collective/src/account.rs [hunk 4]
```diff
@@ -68,6 +68,29 @@ pub struct InitializeCollectiveV0<'info> {
...
+    b"collective", 
+    mint.key().as_ref()],
+    payer=payer,
+    bump,
+    space=312
+  )]
+  pub collective: Box<Account<'info, CollectiveV0>>,
+  pub mint: Box<Account<'info, Mint>>,
+  #[account(
+    constraint = owner_token_ref.authority.unwrap() == payer.key() || mint.mint_authority.unwrap() == payer.key()
```
Comment (ChewingGlass) [inline programs/spl-token-collective/src/account.rs:84]: The mint authority check shouldn't matter right, this endpoint is explicitly for social tokens?

File: programs/spl-token-collective/src/account.rs [hunk 5]
```diff
@@ -68,6 +68,29 @@ pub struct InitializeCollectiveV0<'info> {
...
+  )]
+  pub owner_token_ref: Box<Account<'info, TokenRefV0>>,
```
Comment (ChewingGlass) [inline programs/spl-token-collective/src/account.rs:86]: No seeds check, so this really could be any token_ref. mint token ref, owner token ref, etc. Can probably just call it `token_ref`.

File: programs/spl-token-collective/src/account.rs [hunk 6]
```diff
@@ -68,6 +68,29 @@ pub struct InitializeCollectiveV0<'info> {
...
+pub struct InitializeCollectiveForSocialTokenV0<'info> {
+  #[account(init, seeds = [
+    b"collective", 
+    mint.key().as_ref()],
+    payer=payer,
+    bump,
+    space=312
+  )]
+  pub collective: Box<Account<'info, CollectiveV0>>,
+  pub mint: Box<Account<'info, Mint>>,
```
Comment (ChewingGlass) [inline programs/spl-token-collective/src/account.rs:82]: Missing a check that the mint is the mint that is on the social token

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

File: packages/spl-utils/src/splToken.ts (hunk 3 above)
Comment (ThornM9) [reply packages/spl-utils/src/splToken.ts:17]: Account is deprecated and it was an easy change to switch it out to Keypair in this case

File: programs/spl-token-collective/src/account.rs (hunk 4 above)
Comment (ThornM9) [reply programs/spl-token-collective/src/account.rs:84]: True

**Comment by ChewingGlass** [comment]
@ThornM9 tests are failing

**Comment by ThornM9** [comment]
@ChewingGlass looks like it's good now

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**